### DISCUSSION TITLE
Assembly quality using different k-mer sizes

### DISCUSSION THREAD

**Discussion by ShailNair** [ablab/spades#736] [Q&A]
Hi,
I assembled  pair-end metagenomic reads via spades.  I used two sets of k-mers to evaluate the assembly quality. The raw reads were 250bp long. 

**Assembly-1** with k-mers- **21,33,55,77**
**Aseembly-2** with k-mers- **21,33,55,77,99,127**

Here is the Metaquast statistics:

**Statistics without reference** | **Assembly-1**
-- | --
contigs | 38417
 contigs (>= 0 bp) | 288336
 contigs (>= 1000 bp) | 14500
contigs (>= 5000 bp) | 2978
contigs (>= 10000 bp) | 1471
contigs (>= 25000 bp) | 583
contigs (>= 50000 bp) | 313
Largest contig | 1048960
Total length | 119714142
N50 | 21147
N75 | 2648
L50 | 682
L75 | 5284
GC (%) | 58.3
Mismatches |  
 N's | 0

**Statistics without reference** | **Assembly-2**
-- | --
contigs | 41525
contigs (>= 0 bp) | 132369
contigs (>= 1000 bp) | 16003
 contigs (>= 5000 bp) | 3174
contigs (>= 10000 bp) | 1523
contigs (>= 25000 bp) | 581
 contigs (>= 50000 bp) | 281
Largest contig | 2737835
Total length | 123551515
N50 | 16944
N75 | 2317
L50 | 875
L75 | 6492
GC (%) | 58.34
Mismatches |  
 N's | 0

My confusion is, from these statistics which is the better-looking assembly?.  Though from N-50 **Assembly-1** looks better, it has less number of contigs than assembly 2.   Also, the largest contig in Assembly-1 is smaller than that of Assembly-2. Or do I need to find the orf's, annotate, and then compare the results? (which will take some time as I have many reads)

**Comment by asl** [comment]
Neither N50, nor # of contigs or the length of largest contig is the "quality of assembly" (especially for metagenomes). It's very easy to obtainer longer contigs that contain many problems.

So you'd need to analyze your data further on to decide.

**Comment by ShailNair** [comment]
I see. I will analyze further and see how the differences are reflected in the two assemblies.

Thanks :relaxed:
